### PR TITLE
11L FullGPTQ + XSA-all + BigramHash 3072×112 — val_bpb 1.11564 (1-seed)

### PR DESCRIPTION
**Pull request by AVINASH0052**
## Summary

Single-seed result: **val_bpb 1.11564047** | 15,832,508 bytes | 8×H100 SXM | seed 1337

This submission does not beat the current merged SOTA (1.1147). Submitted as a non-record for technique documentation and reproducibility.

---

## Results

| Seed | val_bpb | val_loss | artifact_bytes |
|------|---------|----------|----------------|
| 1337 | 1.11564047 | 1.88370722 | 15,832,508 |

Post-EMA (before GPTQ): val_bpb 1.1350  
After int6 GPTQ + sliding-window exact eval (stride=64): val_bpb **1.11564047**

---

## Key Techniques

| Technique | Detail |
|-----------|--------|
| Architecture | 11L, 512d, GQA 8H/4KV |
| MLP | 3× width, LeakyReLU(0.5)² |
| Parameter Banking | Shared weight scales across layers |
| Optimizer | Parallel Muon (8-GPU) |
| Attention | XSA on all 11 layers |
| Hash Embedding | BigramHash 3072×112 |
| Pos Encoding | Partial RoPE (16 of 64 dims) |
| LN Scaling | 1/√(L+1) per layer |
| Skip Connections | U-Net style (layer 0↔10, 1↔9, 2↔8) |
| Volume Embed | VE128 on layers 9, 10 |
| SmearGate | Input smearing gate on embedding |
| EMA | α=0.997, tight SWA start at step 6150 |
| QAT | Late QAT enabled at step 6299, scale=0.1498 |
| Quantization | Full Hessian GPTQ int6, AR self-gen calibration (64 seqs × 2048 tokens) |
| Pruning | Selective ±1 pruning (already fits, no pruning needed) |
| Compression | LZMA-9 |
| Eval | Sliding window stride=64 |
| Training | 6891 steps, avg 87.08ms/step, FA3=True |

---

## Reproduction

```bash
pip install flash_attn_3 --no-deps --find-links \
    https://windreamer.github.io/flash-attention3-wheels/cu128_torch291/
cd records/track_10min_16mb/2026-04-08_11L_FullGPTQ_XSA11_BigramHash3072
SEED=1337 bash run_leaderboard_8xh100.sh

### CODE REVIEW

**Comment by MatoTeziTanka** [comment]
### Community Review — Non-record: 11L FullGPTQ + XSA-all + BigramHash 3072×112 — val_bpb 1.11564 (1-seed)

**BPB:** 1.11564 | **Compliance:** LOOKS CLEAN — pure-neural submission, no TTT/SLOT/n-gram-cache

**What I found in the code** (head SHA `68596decbe06`, file `records/track_10min_16mb/2026-04-08_10L_LeakyReLU2_MLP3x_EMA_Int6QAT/train_gpt.py`):

Static code review found no TTT adaptation function, no SLOT optimization loop, no n-gram-cache class, and no pre-quant val-token fine-tune. The eval path uses the standard sliding-window stride-64 pattern. The submission is a pure-neural architecture iteration on the standard SP1024/SP4096/SP8192 baseline.

**CPU smoke test (CT2038 proteus-engine, 2026-04-11):** import OK in 0.09s, dim=512, layers=9, vocab=1024, code=47545 B, SMOKE_TEST_PASS

**Verdict:** LOOKS CLEAN.

**Recommendation to @cocohearts @valerio-oai @0hq @yuzhougu-oai @notapplica:** **MERGE** pending the usual record-track checks (3-seed validation, under-16MB artifact cap, ≤600s train + ≤600s eval on 8×H100 SXM). No compliance flags from the classification pass — this looks like a clean pure-neural iteration on the standard baseline.

*Auto-classification caveat:* this review was drafted by the AST-based classifier. If there's a non-standard eval mechanism (logit postprocessing, hedge mixing, etc.) that I missed because it's factored into a helper file or a non-standard function name, please flag it and I'll re-run the audit manually.

---
*Reviewed by [@MatoTeziTanka](https://github.com/MatoTeziTanka) — [The Agora](https://matotezitanka.github.io/parameter-golf). CPU smoke test (CT2038 proteus-engine, 2026-04-11): import OK in 0.09s, dim=512, layers=9, vocab=1024, code=47545 B, SMOKE_TEST_PASS. Classification via deterministic AST-based `classify_prs.py` (pattern bank derived from ~65 manually-reviewed PRs earlier in the 2026-04-11 sweep). This review was auto-drafted from a template and spot-checked before posting — if the template misread your code, please call it out so I can iterate the classifier.*

**Comment by AVINASH0052** [comment]
Hi @MatoTeziTanka — thanks for the review!

One correction: the file audited appears to be [train_gpt.py](vscode-file://vscode-app/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-browser/workbench/workbench.html) (an earlier experiment that was accidentally included in this PR). The smoke test showing layers=9, code=47545B confirms this — that's the old 10L file with a NUM_LAYERS default bug.

The actual submission for this PR is:
[train_gpt.py](vscode-file://vscode-app/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-browser/workbench/workbench.html)
— 11 layers, 512d, GQA 8H/4KV, full Hessian GPTQ int6, XSA-all, BigramHash 3072×112.
Real leaderboard run: layers=11, val_bpb=1.11564047, 15,832,508 bytes PASS.

Also: this is a non-record submission (val_bpb 1.11564 does not beat merged SOTA 1.1147), so the 3-seed validation requirement should not apply here.
